### DISCUSSION TITLE
How to transform EDI / X12 to an Encoded URL string

### DISCUSSION THREAD

**Discussion by tcotugno** [nextgenhealthcare/connect#6029] [Q&A]
Hi, pretty green here and trying to find a way to transform an X12 to and encoded URL string.  I see Mirth has some canned templates to convert EDI to XML but not this.  Thanks in advance

**Comment by pacmano1** [comment]
Well, you are only asking "I have a string and I need to URL encode it".   If you are talking about changing x12 to some other format that is a different question.

If you are talking about doing an http post with the x12 as part of a form or the body, that is yet another question.

So, maybe proivde a bit more?

**Reply by tcotugno** [reply]
Thanks, I am trying to transform an X12 message to a URL encoded format.  And place this formatted message into the URL of the HTTP destination.

For example transforming this:

`*SOMETHING     *231120*0912*^*00501*000113999*0*P*:~`

to 

`%2ASOMETHING%20%20%20%20%20%2A231120%2A0912%2A%5E%2A00501%2A000113999%2A0%2AP%2A%3A~`

So i think I need to take the variable:${message.rawData} as the input and then transform to and use this as the output: ${message.encodedData}

**Reply by pacmano1** [reply]
in a transformer hit ctrl-space.

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/4c06a35e-4355-4395-b250-92b046095d2c)

**Reply by tcotugno** [reply]
Thank you.  Seeing an error that it cant find the function?

218: //Convert Raw X12 to Encoded URL
219: SerializerFactory.getSerializer('EDI/X12').encodeURI(message);
222: //encodeURI(arg0)

LINE NUMBER:	219
DETAILS:	TypeError: Cannot find function encodeURI in object com.mirth.connect.plugins.datatypes.edi.EDISerializer@67fee84e.

**Reply by pacmano1** [reply]
It takes a string. you are passing an object.

**Reply by tcotugno** [reply]
Thanks!!
